build.assets/notarize-binaries.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+set -e
+
+usage() {
+  echo "Notarizes MacOS binaries with xcrun."
+  echo "Usage: $(basename $0) [file ...]" 1>&2
+  exit 1
+}
+
+# Pulled from https://stackoverflow.com/a/26809278
+shell_array_to_json_array() {
+    echo -n '['
+    while [ $# -gt 0 ]; do
+        x=${1//\\/\\\\}
+        echo -n \"${x//\"/\\\"}\"
+        [ $# -gt 1 ] && echo -n ', '
+        shift
+    done
+    echo ']'
+}
+
+# This is largely pulled from `build-common.sh` but modified for this use case
+sign_and_notarize_binaries() {
+  local bundle_id="$1"
+  local targets=(${@:2})
+
+  local notarization_zip="teleport.zip"
```

Comment:
I thought you'd mentioned that zipping files is no longer required. Is this still needed then?

Reply:
It's still required that we zip files for within the scope of this tool. That being said, the zipped file is _not_ used outside of this function, uploaded, or published. The `gon` tool simply requires that internally the binaries be archived when uploading to Apple.

Just to reiterate and clarify, we are still publishing `*.tar.gz` files, and not publishing `*.zip` files. This is only used internally in the build step.